libs/video-view/src/index.tsx

```diff
@@ -30,22 +34,48 @@ const VideoView = ({
   displayVideo = true,
   displayFullscreenButton = true,
   mediaStream,
+  url,
   // statistics,
   label,
   placeholderNode,
   onClick,
   showDotIndicator,
+  volume = 1,
+  setMediaStream,
+  removeMediaStream,
 }: VideoViewProps) => {
   const videoRef = useRef<HTMLVideoElement>(null);
   const [isFullScreen, setIsFullScreen] = useState(false);
   const [loadingVideo, setLoadingVideo] = useState(true);
   // const [showStatisticsInfo, setShowStatisticsInfo] = useState(false);
 
   useEffect(() => {
+    let stream: MediaStream;
     if (videoRef.current) {
-      videoRef.current.srcObject = mediaStream ?? null;
+      if (url) {
+        console.log('trigger');
+        //@ts-expect-error property exists but it isn't in the built-in type
+        stream = videoRef.current?.captureStream() as MediaStream;
```

Comment:
we'd better use `video` HTMLVideoElement's `canplay` event callback to trigger user's callback `onSrcMediaStreamReady`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canplay_event